spec/controllers/articles_controller_spec.rb

```diff
@@ -111,4 +111,25 @@
     end
   end
 
+  describe 'GET #index' do
+    context 'when user is logged in' do
+      let(:user) { create(:user) }
+
+      before(:each) do
+        allow(controller).to receive(:current_user).and_return(user)
+      end
+      it 'renders the index template' do
+        get :index
+        expect(response).to render_template "index"
+      end
+    end
+
+
+    context 'when user is not logged in' do
+      it 'renders the index template' do
+        get :index
+        expect(response).to render_template "index"
+      end
+    end
+  end
```

Comment:
Both scenarios are identical as they should, because that's what the story asks for, but what are they actually asserting?

You are asserting that if the `articles_controller#index` action renders the `index` template. But have you written the code responsible for this or is this an innate Rails feature?

If we go back to story, it says that we should list the articles, but is that what you are asserting? Try this, edit the `views/articles/index.html.erb` and remove the loop to list the articles, hence making the page blank, and run the tests. They will pass, although the articles are not being displayed. How can you modify your tests to assure they would fail in this scenario?

Reply:
Actually, to make my point more clear, the above comment addresses the `view` which does not have any tests yet, thus instead of modifying the view file, make a change to the controller, commenting out this line `# @articles = Article.order('created_at DESC')` and run the tests.